slam_vdb/CMakeLists.txt

```diff
@@ -0,0 +1,203 @@
+endif()
+
+# find all dependencies
+find_package(Eigen3 REQUIRED)
+
+# -----------------------------------------------------------------------
+# Options
+# Set -DBUILD_TESTING=ON for compiling unit tests
+# Set -DCODE_COVERAGE=ON to generate a code coverage reports when unit tests are executed
+# Set -DBUILD_BENCHMARKS=ON for compiling benchmarks files
+# Set -DBUILD_DOCUMENTATION=ON to create docs target. Doxygen documentation can be generated with "make docs"
+# -----------------------------------------------------------------------
+# Build dynamically linkable libraries
+unset(BUILD_SHARED_LIBS)
+option(BUILD_SHARED_LIBS "Build sharedlibs" ON)
+message(STATUS "Build sharedlibs: " ${BUILD_SHARED_LIBS})
+
+# Build unit test files - Deafult is OFF
+unset(BUILD_TESTING)
+option(BUILD_TESTING "Build testing" OFF)
+message(STATUS "Build testing: " ${BUILD_TESTING})
+
+# Build code coverage files - Deafult is OFF
+unset(CODE_COVERAGE)
+option(CODE_COVERAGE "Enable coverage reporting" OFF)
+message(STATUS "Enable coverage reporting: " ${CODE_COVERAGE})
+
+# Build code benchmark files - Deafult is OFF
+unset(BUILD_BENCHMARK)
+option(BUILD_BENCHMARK "Build benchmark" OFF)
+message(STATUS "Build benchmark: " ${BUILD_BENCHMARK})
+
+# Build doxygen target to create docs - Deafult is OFF
+unset(BUILD_DOCUMENTATION)
+option(BUILD_DOCUMENTATION "Build Documentation" OFF)
+message(STATUS "Build Documentation: " ${BUILD_DOCUMENTATION})
+
+# -----------------------------------------------------------------------
+# Target
+# -----------------------------------------------------------------------
+# If individual libraries have different versions  - set them in set_target_properties
+# Target linked with interface
+# -----------------------------------------------------------------------
+# slam_vdb Target
+# -----------------------------------------------------------------------
+add_library(slam_vdb_interface INTERFACE)
+target_compile_features(slam_vdb_interface INTERFACE cxx_std_17)
+target_compile_options(slam_vdb_interface INTERFACE
+	"$<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>:-ffp-contract=off>"
+	"$<$<CXX_COMPILER_ID:GNU>:-ffp-contract=off>"
+	"$<$<CXX_COMPILER_ID:MSVC>:/fp:precise>"
+)
+add_library(slam_vdb "")
+add_library(SLAM_VDB::slam_vdb ALIAS slam_vdb)
+
+target_link_libraries(slam_vdb
+	PUBLIC slam_vdb_interface
+	INTERFACE Eigen3::Eigen
+)
+
+# Add the headers to the target
+file(GLOB_RECURSE _headers ${CMAKE_CURRENT_LIST_DIR}/include CONFIGURE_DEPEND)
+target_sources(slam_vdb PRIVATE ${_headers})
+unset(_headers)
+
+target_include_directories(slam_vdb
+	PUBLIC
+	$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+	$<BUILD_INTERFACE:${EIGEN3_INCLUDE_DIRS}>
+	$<INSTALL_INTERFACE:include>)
+
+# Add the sources to the target
+file(GLOB_RECURSE _sources ${CMAKE_CURRENT_LIST_DIR}/src/*.cpp CONFIGURE_DEPEND)
+target_sources(slam_vdb PRIVATE ${_sources})
+unset(_sources)
```

Comment:
I believe we should be able to create a header-only library and able to link it with ROS2 wrapper side (meaning we dont necessarily need to create a src folder and create a class there). I found [this blog post](https://ryandeangraham.medium.com/building-a-header-only-library-hello-world-f9dd05ac1066) to be really useful for doing this part.

Reply:
This configuration already follows the procedure described in the blog post. Just that you have an `.so` file installed along with headers